src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -117,6 +113,10 @@ class instance_scarlet_monastery : public InstanceMapScript
                             HandleGameObject(PumpkinShrineGUID, false);
                         }
                         break;
+                    case DATA_MOGRAINE_AND_WHITE_EVENT:
+                        if (state == DONE || state == IN_PROGRESS)
+                            HandleGameObject(HighInquisitorsDoorGUID, true);
```

Comment:
why you are not using generic doors handling?

Reply:
Generic door handling is what made it stay open all the time. I'm not sure how it should be set up for a door that should only open mid-fight.

Reply:
By using DOOR_TYPE_SPAWN_HOLE instead of DOOR_TYPE_ROOM

Reply:
If it should open mid-fight and stay open after encounter then it needs manual handling

Reply:
@Wyreth 
For the Door handling eventual take a look in the cMangos code.
In Cmangos, the doors work correctly.